Description: Make renegotiation work for TLS 1.2, 1.1 by not using a lower
 record version client hello workaround if renegotiating.
Origin: upstream, http://cvs.openssl.org/chngview?cn=22567
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/openssl/+bug/1020621

Index: openssl-1.0.1/ssl/s3_pkt.c
===================================================================
--- openssl-1.0.1.orig/ssl/s3_pkt.c	2012-07-03 11:35:05.605494750 -0400
+++ openssl-1.0.1/ssl/s3_pkt.c	2012-07-03 11:35:44.529495746 -0400
@@ -740,6 +740,7 @@
 	 * bytes and record version number > TLS 1.0
 	 */
 	if (s->state == SSL3_ST_CW_CLNT_HELLO_B
+				&& !s->renegotiate
 				&& TLS1_get_version(s) > TLS1_VERSION)
 		*(p++) = 0x1;
 	else
